src/viewer.js

```diff
@@ -411,17 +411,21 @@ const Viewer = function Viewer(targetOption, options = {}) {
         const featureId = urlParams.feature;
         const layerName = featureId.split('.')[0];
         const layer = getLayer(layerName);
+        const clusterSource = layer.getSource().source;
+
         if (layer) {
           layer.once('postrender', () => {
             let feature;
             const type = layer.get('type');
-            feature = layer.getSource().getFeatureById(featureId);
-            if (type === 'WFS') {
+            if (clusterSource && type === 'WFS') {
+              feature = clusterSource.getFeatureById(featureId);
+            } else if (type === 'WFS') {
               feature = layer.getSource().getFeatureById(featureId);
             } else {
               const id = featureId.split('.')[1];
               let origin = layer.getSource();
               feature = origin.getFeatureById(id);
+              feature = layer.getSource().getFeatureById(featureId);
```

Comment:
I think this row just messes things up.

Reply:
Refactored this and came up with that topojson has problems with this feature
             `if (type === 'WFS' && clusterSource) {
              feature = clusterSource.getFeatureById(featureId);
              console.log(feature)

            } else if (type === 'WFS') {
              feature = layer.getSource().getFeatureById(featureId);
              console.log(feature)

            } else if (type === 'GEOJSON' && clusterSource) {
              feature = clusterSource.getFeatureById(id);
              console.log(feature)

            } else if (type === 'GEOJSON') {
              feature = layer.getSource().getFeatureById(id);
              console.log(feature)

            } else if (type === 'TOPOJSON' && clusterSource) {
              feature = clusterSource.getFeatureById(id);
              console.log(feature)

            } else if (type === 'TOPOJSON') {
              feature = layer.getSource().getFeatureById(id);
              console.log(feature)
            }`

Reply:
Is that a working solution?

Reply:
This is an example where WFS and GEOJSON is returning a feature but TOPOJSON returning null. Not sure why, the getFeatureById is pretty straight forward..

Reply:
could it be your layer source?

Reply:
Yes, Im confused where in the prototype the feature is found..

Reply:
I tried the socken.json topojson file and that file seems to lack IDs. The id in the permalink is perhaps the OL id.

Reply:
If i share Övik (id 16) in geojson. Permalink is referencing origo-cities.7. I **thought** getFeatureById was straight forward, I was wrong.
Testing on fme converted origo-cities.geojson -> origo-cities.json and in topojson i get id 94

Reply:
So, topojson object id should not be nested inside properties. There is nothing wrong with the code as far as I can se.